src/Microsoft.Azure.WebJobs.Extensions.EventHubs/PartitionedSender.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.EventHubs;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.EventHubs
+{
+    /// <summary>
+    /// Send events with partition keys to EventHub.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class PartitionedSender
```

Comment:
Should we introduce an interface similar to IAsyncCollector and have customers use that instead of this concrete type? Call it IPartitionedAsyncCollector maybe?